llvm/lib/Object/COFFObjectFile.cpp

```diff
@@ -800,6 +800,218 @@ Error COFFObjectFile::initLoadConfigPtr() {
     }
   }
 
+  // Interpret and validate dynamic relocations.
+  uint32_t DynamicRelocTableOffset = 0, DynamicRelocTableSection = 0;
+  if (is64()) {
+    auto Config = getLoadConfig64();
+    if (Config->Size >=
+        offsetof(coff_load_configuration64, DynamicValueRelocTableSection) +
+            sizeof(Config->DynamicValueRelocTableSection)) {
+      DynamicRelocTableSection = Config->DynamicValueRelocTableSection;
+      DynamicRelocTableOffset = Config->DynamicValueRelocTableOffset;
+    }
+  } else {
+    auto Config = getLoadConfig32();
+    if (Config->Size >=
+        offsetof(coff_load_configuration32, DynamicValueRelocTableSection) +
+            sizeof(Config->DynamicValueRelocTableSection)) {
+      DynamicRelocTableSection = Config->DynamicValueRelocTableSection;
+      DynamicRelocTableOffset = Config->DynamicValueRelocTableOffset;
+    }
+  }
+
+  Expected<const coff_section *> Section = getSection(DynamicRelocTableSection);
+  if (!Section)
+    return Section.takeError();
+  if (*Section) {
+    ArrayRef<uint8_t> Contents;
+    if (Error E = getSectionContents(*Section, Contents))
+      return E;
+
+    Contents = Contents.drop_front(DynamicRelocTableOffset);
+    if (Contents.size() < sizeof(coff_dynamic_reloc_table))
+      return createStringError(object_error::parse_failed,
+                               "Too large DynamicValueRelocTableOffset (" +
+                                   Twine(DynamicRelocTableOffset) + ")");
+
+    DynamicRelocTable =
+        reinterpret_cast<const coff_dynamic_reloc_table *>(Contents.data());
+
+    if (DynamicRelocTable->Version != 1 && DynamicRelocTable->Version != 2) {
+      DynamicRelocTable = nullptr;
+      return Error::success();
```

Comment:
Why is this not an error?

Reply:
I meant it for forward-compatibility. If there is a newer version in the future, we won't be able to read it, but the object could be otherwise still useful.

Reply:
For objdump, I can see an argument for continuing to dump (ideally with a warning), but other uses of object files likely want to bail out of they can't parse an important part of the object file.  And the API doesn't really provide any way to tell what happened.

Reply:
I pushed a version that makes it an error, thanks.

Reply:
FWIW, this distinction between verifying all aspects, or just reading as much as possible with as little judgement as possible, is indeed quite frustrating at times.

If you have an object file that our tools reject, our tools also are quite unfriendly in helping you figure out what really is wrong. (I remember doing a WIP/throwaway branch sometime to make all the COFFObjectFile errors much more verbose, and to continue parsing as far as possible, to let me analyze some file.)